tests/integration/security/authorization_test.go

```diff
@@ -512,7 +512,210 @@ func TestAuthorization_IngressGateway(t *testing.T) {
 
 			for _, tc := range cases {
 				ctx.NewSubTest(tc.Name).Run(func(ctx framework.TestContext) {
-					authn.CheckIngressOrFail(ctx, ingr, tc.Host, tc.Path, "", tc.WantCode)
+					authn.CheckIngressOrFail(ctx, ingr, tc.Host, tc.Path, nil, "", tc.WantCode)
+				})
+			}
+		})
+}
+
+// TestAuthorization_IngressGatewayRemoteIp tests the authorization policy on ingress gateway with remote ip.
+func TestAuthorization_IngressGatewayRemoteIp(t *testing.T) {
```

Comment:
Can we merge the 3 tests to the existing TestAuthorization_IngressGateway?

You can just add more rules for the remoteIP, notRemoteIP and remote.ip attribute to the AuthorizationPolicy in the `v1beta1-ingress-gateway.yaml.tmpl` and update the `cases` to support the extra header configuration.

This not only remove a lot duplicate code but also increase the test speed a lot (each test function will create and deploy its workload which is time consuming and we should try to reuse).